x-pack/plugins/cases/server/client/user_actions/users.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { GetCaseUsersResponse } from '../../../common/api';
+import { GetCaseUsersResponseRt } from '../../../common/api';
+import type { OwnerEntity } from '../../authorization';
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+import type { CasesClient } from '../client';
+import type { CasesClientArgs } from '../types';
+import type { GetUsersRequest } from './types';
+
+export const getUsers = async (
+  { caseId }: GetUsersRequest,
+  casesClient: CasesClient,
+  clientArgs: CasesClientArgs
+): Promise<GetCaseUsersResponse> => {
+  const {
+    services: { userActionService },
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const entities: OwnerEntity[] = [];
+    const participants: GetCaseUsersResponse['participants'] = [];
+
+    // ensure that we have authorization for reading the case
+    await casesClient.cases.resolve({ id: caseId, includeComments: false });
+    const { participants: participantsInfo, users: userProfileUids } =
+      await userActionService.getUsers({ caseId });
+
+    for (const participant of participantsInfo) {
+      entities.push({ id: participant.id, owner: participant.owner });
```

Comment:
Hey Antonio! You are right that we should authorize the user action SO. The `participant.owner` is the owner of the user action SO. The `participantsInfo` contains all SO returned by the aggregation. I named the participants because is the SO that contains the participant I want to return. I can rename it to something clearer. Any suggestions?

Reply:
Ah, the user action is actually in the participants' portion of the aggregation as defined by `buildParticipantsAgg` right? You mentioned offline having added the `docs` portion for that.

I don't really have a better naming suggestion but now things make sense for me, thanks 👌